pkg/workloadmeta/collectors/internal/process/process.go

```diff
@@ -0,0 +1,145 @@
+	workloadmetaExtractor "github.com/DataDog/datadog-agent/pkg/process/metadata/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+)
+
+const collectorId = "local-process"
+
+// Used for testing
+var c *collector
+
+func init() {
+	// The process collector can run either when workloadmeta is in local or remote collector mode.
+	workloadmeta.RegisterCollector(collectorId, newProcessCollector)
+	workloadmeta.RegisterRemoteCollector(collectorId, newProcessCollector)
+}
+
+func newProcessCollector() workloadmeta.Collector {
+	// TODO: Inject config.Datadog via fx once collectors are migrated to components.
+	ddConfig := config.Datadog
+
+	wlmExtractor := workloadmetaExtractor.NewWorkloadMetaExtractor(ddConfig)
+
+	processData := checks.NewProcessData(ddConfig)
+	processData.Register(wlmExtractor)
+
+	c = &collector{
+		ddConfig:        ddConfig,
+		wlmExtractor:    wlmExtractor,
+		grpcServer:      workloadmetaExtractor.NewGRPCServer(ddConfig, wlmExtractor),
+		processData:     processData,
+		collectionClock: clock.New(),
+		pidToCid:        make(map[int]string),
+	}
+	return c
+}
+
+// Compile time check to ensure that `collector` implements `workloadmeta.Collector`.
+var _ workloadmeta.Collector = (*collector)(nil)
+
+type collector struct {
+	ddConfig config.ConfigReader
+
+	processData *checks.ProcessData
+
+	wlmExtractor *workloadmetaExtractor.WorkloadMetaExtractor
+	grpcServer   *workloadmetaExtractor.GRPCServer
+
+	pidToCid map[int]string
+
+	collectionClock clock.Clock
+}
+
+func (c *collector) Start(ctx context.Context, store workloadmeta.Store) error {
+	if enabled, err := Enabled(c.ddConfig); !enabled {
+		return err
+	}
+
+	err := c.grpcServer.Start()
+	if err != nil {
+		return err
+	}
+
+	collectionTicker := c.collectionClock.Ticker(
+		c.ddConfig.GetDuration("workloadmeta.local_process_collector.collection_interval"),
+	)
+
+	filter := workloadmeta.NewFilter([]workloadmeta.Kind{workloadmeta.KindContainer}, workloadmeta.SourceAll, workloadmeta.EventTypeAll)
+	containerEvt := store.Subscribe(collectorId, workloadmeta.NormalPriority, filter)
+
+	go c.run(ctx, store, containerEvt, collectionTicker)
+
+	return nil
+}
+
+func (c *collector) run(ctx context.Context, store workloadmeta.Store, containerEvt chan workloadmeta.EventBundle, collectionTicker *clock.Ticker) {
+	defer c.grpcServer.Stop()
+	defer store.Unsubscribe(containerEvt)
+	defer collectionTicker.Stop()
+
+	for {
+		select {
+		case evt := <-containerEvt:
+			c.handleContainerEvent(evt)
+		case <-collectionTicker.C:
+			err := c.processData.Fetch()
+			if err != nil {
+				_ = log.Error("Error fetching process data:", err)
+			}
+		case <-ctx.Done():
+			log.Infof("The %s collector has stopped", collectorId)
+			return
+		}
+	}
+}
+
+// Pull is unused at the moment used due to the short frequency in which it is called.
+// In the future, we should use it to locally in workload-meta.
```

Comment:
```suggestion
// In the future, we should use it locally in workload-meta.
```

Reply:
This was an unfinished sentence. I changed it to:

>In the future, we should use it to poll for processes that have been collected and store them in workload-meta.